relay-server/src/services/processor/event.rs

```diff
@@ -145,10 +149,37 @@ pub fn extract<Group: EventProcessing>(
 
     metrics.bytes_ingested_event = Annotated::new(event_len as u64);
 
+    if let Some(event_value) = event.value() {
+        let convert_breadcrumbs_to_logs = utils::sample(
+            global_config
+                .options
+                .ourlogs_breadcrumb_extraction_sample_rate
+                .unwrap_or(0.0),
+        );
+
+        if convert_breadcrumbs_to_logs {
+            relay_log::trace!("extracting breadcrumbs to logs");
+            let ourlogs: Vec<relay_event_schema::protocol::OurLog> =
+                breadcrumbs_to_ourlogs(event_value, config.max_breadcrumbs_converted());
+
+            if !ourlogs.is_empty() {
+                for ourlog in ourlogs {
+                    let mut log_item = Item::new(ItemType::Log);
+                    if let Ok(payload) = Annotated::new(ourlog).to_json() {
+                        log_item.set_payload(ContentType::Json, payload);
+                        envelope.add_item(log_item);
```

Comment:
I was considering having this in `processor/ourlog.rs` as an `extract_...` function called in both `process_errors` and `process_transactions` instead, thoughts?

Reply:
Yes, I would definitely move this out of `extract` and into a separate call in `process_errors`. We can guard it with `if_processing!` to make sure we only extract in the innermost relay. That saves network bandwidth and makes it easier to iterate without having to worry about outdated customer Relay's logic.